### PR TITLE
Adding implementation for PW-7280

### PR DESCRIPTION
**Pull request by zenit2001**
This PR contains the implementation of uploading images in Adyen Giving section.
Also, it changes the textual description for POS section.

### CODE REVIEW

File: src/cartridges/bm_adyen/cartridge/static/default/adyenSettings.js [hunk 1]
```diff
@@ -86,6 +90,76 @@ document.addEventListener('DOMContentLoaded', () => {
     this.classList.toggle('bi-eye');
   }
 
+  // open Adyen Giving Background upload page
+  function uploadAdyenGivingBackground() {
+    const params = 'resizable=no,width=1000,height=500,left=100,top=100';
```
Comment (wboereboom) [inline src/cartridges/bm_adyen/cartridge/static/default/adyenSettings.js:95]: Does this work for all screen sizes?

File: src/cartridges/bm_adyen/cartridge/static/default/adyenSettings.js [hunk 2]
```diff
@@ -86,6 +90,76 @@ document.addEventListener('DOMContentLoaded', () => {
...
+    const openedWindow = window.open(
+      window.adyenGivingBackgroundUrl,
+      'test',
```
Comment (wboereboom) [inline src/cartridges/bm_adyen/cartridge/static/default/adyenSettings.js:98]: Why is this called test?

File: src/cartridges/bm_adyen/cartridge/static/default/adyenSettings.js [hunk 3]
```diff
@@ -86,6 +90,76 @@ document.addEventListener('DOMContentLoaded', () => {
...
+      params,
+    );
+    const loop = setInterval(() => {
```
Comment (wboereboom) [inline src/cartridges/bm_adyen/cartridge/static/default/adyenSettings.js:101]: Have you tried the `onbeforeunload ` event for the window? If that does not work this seems to be the go-to solution

File: src/cartridges/bm_adyen/cartridge/static/default/adyenSettings.js [hunk 4]
```diff
@@ -86,6 +90,76 @@ document.addEventListener('DOMContentLoaded', () => {
...
+      if (openedWindow.closed) {
+        window.location.reload();
+        clearInterval(loop);
+      }
+    }, 1000);
+  }
+
+  // open Adyen Giving Logo upload page
+  function uploadAdyenGivingLogo() {
+    const params = 'resizable=no,width=1000,height=500,left=100,top=100';
```
Comment (wboereboom) [inline src/cartridges/bm_adyen/cartridge/static/default/adyenSettings.js:111]: Duplicate code

File: src/cartridges/bm_adyen/cartridge/static/default/adyenSettings.js [hunk 5]
```diff
@@ -86,6 +90,76 @@ document.addEventListener('DOMContentLoaded', () => {
...
+    const parts = imageUrl.split('/');
+    const imageName = parts.pop();
+    return imageName;
+  }
+
+  function printBackgroundImageName() {
+    document.getElementById('backgroundFileName').innerHTML = '';
+    const ul = document.getElementById('backgroundFileName');
+    const nameOfImage = getImageName(window.backgroundValueField);
+    if (nameOfImage?.length > 0) {
```
Comment (wboereboom) [inline src/cartridges/bm_adyen/cartridge/static/default/adyenSettings.js:135]: the contents of this if block are a bit difficult to parse.
Can you maybe change some variable names or add some comments to make it more readable?

File: src/cartridges/bm_adyen/cartridge/static/default/adyenSettings.js [hunk 6]
```diff
@@ -86,6 +90,76 @@ document.addEventListener('DOMContentLoaded', () => {
...
+      li.appendChild(text);
+      ul.appendChild(li);
+    }
+  }
+
+  function printLogoImageName() {
+    document.getElementById('logoFileName').innerHTML = '';
+    const ul = document.getElementById('logoFileName');
+    const nameOfImage = getImageName(window.logoValueField);
+    if (nameOfImage?.length > 0) {
```
Comment (wboereboom) [inline src/cartridges/bm_adyen/cartridge/static/default/adyenSettings.js:151]: duplicate code

File: src/cartridges/bm_adyen/cartridge/templates/default/adyenSettings/settingCards/adyenGivingSettings.isml [hunk 7]
```diff
@@ -63,31 +68,36 @@
                <label class="form-title mb-0" for="backgroundFile">Background image for Adyen Giving component</label>
                <small id="backgroundImageHelo" class="form-text mb-1">Go to the classic page view on the top of this section to set up your Adyen Giving and upload the image there.<br>
                Once you save the configuration, the image will be automatically updated here.</small>
-               <div class="input-fields">
-                  <input type="text" class="form-control" name="AdyenGiving_backgroundUrl" id="backgroundFile" value="${AdyenConfigs.getAdyenGivingBackgroundUrl() || ''}" disabled>
+               <script>
+                  //Token and path can be found in settings.isml file
+                  window.adyenGivingBackgroundUrl = background_token +"#" + addition1 + addition4 + addition5;
```
Comment (wboereboom) [inline src/cartridges/bm_adyen/cartridge/templates/default/adyenSettings/settingCards/adyenGivingSettings.isml:73]: Can we move this to adyenSettings.js?

File: src/cartridges/bm_adyen/cartridge/templates/default/adyenSettings/settingCards/adyenGivingSettings.isml [hunk 8]
```diff
@@ -63,31 +68,36 @@
...
             </div>
             <div class="form-group">
                <label class="form-title mb-0" for="logoFile">Logo in Adyen Giving component</label>
                <small id="backgroundImageHelo" class="form-text mb-1">Go to the classic page view on the top of this section to set up your Adyen Giving and upload the image there.<br>
                Once you save the configuration, the image will be automatically updated here.</small>
-               <div class="input-fields">
-                  <input type="text" class="form-control" name="AdyenGiving_logoUrl" id="logoFile" value="${AdyenConfigs.getAdyenGivingLogoUrl() || ''}" disabled>
+               <script>
+                  //Token and path can be found in settings.isml file
+                  window.adyenGivingLogoUrl = logo_token +"#" + addition1 + addition4 + addition6;
```
Comment (wboereboom) [inline src/cartridges/bm_adyen/cartridge/templates/default/adyenSettings/settingCards/adyenGivingSettings.isml:90]: Can we join the different script tags that are scattered throughout this template?

File: src/cartridges/bm_adyen/cartridge/templates/default/adyenSettings/settings.isml [hunk 9]
```diff
@@ -9,10 +9,15 @@
       <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/js/bootstrap.bundle.min.js" integrity="sha384-MrcW6ZMFYlzcLA8Nl+NtUVF0sA7MsXsP1UyJoMp4YLEuNSfAP+JcXn/tWtIaxVXM" crossorigin="anonymous"></script>
       <meta name="viewport" content="width=device-width, initial-scale=1">
       <script>
-         let token = "${URLUtils.https('ViewApplication-BM')}?csrf_token=${dw.web.CSRFProtection.generateToken()}"
+         const token = "${URLUtils.https('ViewApplication-BM')}?csrf_token=${dw.web.CSRFProtection.generateToken()}"
+         const logo_token = "${URLUtils.https('ViewApplication-BM')}?csrf_token=${dw.web.CSRFProtection.generateToken()}"
+         const background_token = "${URLUtils.https('ViewApplication-BM')}?csrf_token=${dw.web.CSRFProtection.generateToken()}"
          const addition1 = "/?preference";
```
Comment (wboereboom) [inline src/cartridges/bm_adyen/cartridge/templates/default/adyenSettings/settings.isml:15]: These names are not very descriptive.  what are addition1 through 6? 
Consider changing the names or add some comments.

**Review by wboereboom — CHANGES_REQUESTED** [review]

File: src/cartridges/bm_adyen/cartridge/static/default/adyenSettings.js (hunk 2 above)
Comment (zenit2001) [reply src/cartridges/bm_adyen/cartridge/static/default/adyenSettings.js:98]: It is the popup window name, forgot to change it.
Will update!

**Comment by amihajlovski** [comment]
One comment regarding the commit messages. Instead of having `fixed comments`, `resolved feedback` or `fixed one mistake` we should have more descriptive message with exact change that happened like `renamed element id and removed one html block connected to it`. This way any one who opens the git log will have clear vision of what are the changes :)

**Review by wboereboom — APPROVED** [review]

**Review by zaiddreakh — APPROVED** [review]

**Pull request merged**